### PR TITLE
Added Policy Documentation to the Repository

### PR DESCRIPTION
**Pull request by mitchelsellers**
## Summary
Added updated Markdown files, and added links from the main Readme.md file to document the new policies and procedures mutually agreed upon by the Community and ESW relating to the future developments and releases for the DNN Platform.

### CODE REVIEW

File: .github/PULL_REQUEST_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+## Prerequisites
+The following items must all be true prior to a pull request being submitted to the DNN_Platform, if any of the following items are not true the pull request will be returned for edits referencing the specific missing item(s).
+
+* Change should represent an entire solution for the issue at hand.  Partial requests will NOT be processed.
+* Change should have a supporting issue logged on the DNN_Platform GitHub account, documenting the issue resolved, following the procedures outlined on the [Contribute Page](CONTRIBUTING.md)
+* Exception: Security items can be addressed individually, and should be initially communicated to the security@dnnsoftware.com email address for coordination
+* Pull request comment should contain at a minimum the following details (When creating, a default template will also prompt you for the proper information)
+--* Issue #, referenced such as “Fixes #24”
+--* Release Note: Suggested release note for the change such as “Improvement of user profile to improve performance when searching.”
+--* Testing Steps: Suggested testing steps for validation of the change
```
Comment (tpluscode) [inline .github/PULL_REQUEST_PROCESS.md]: Let's take the occasion to add *Testing steps* to issue template and *Fixes #XXX* to the PR template

File: .github/PULL_REQUEST_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Community review of submitted pull requests is encouraged, and all pull requests must be reviewed by at least **two (2)** community approvers before the change can be approved.  Once approved by two unique individuals the change can be merged by the secondary approval.  At the current time the following community members are designated approvers.
+
+* Mitchel Sellers ([mitchelsellers](https://github.com/mitchelsellers)) - Community Technology Advisory Group Lead
+* Oliver Hine ([ohine](https://github.com/ohine))
+* Brian Dukes ([bdukes](https://github.com/bdukes))
+* Peter Donker ([donker](https://github.com/donker)) - Community Developer Advisory Group Lead
+* Vicnec Masanas ([vmasanas](https://github.com/vmasanas))
+* Erik van Ballegoij ([erikvb](https://github.com/erikvb))
+
+An individual performing the code review should validate at a minimum the following.
```
Comment (tpluscode) [inline .github/PULL_REQUEST_PROCESS.md]: Do you want to mention that tests are greatly appreciated?

File: .github/PULL_REQUEST_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# DNN Platform Pull Request Proceseses
+
+The goal of this document is to set standards for the review, processing, and approval of all Pull requests into the DNN_Platform GitHub repository, as well as associated commmunity open-source repositories.  These rules apply to ALL pull requests, regardless of the submitter, or complexity of the change.
+
+## Prerequisites
+The following items must all be true prior to a pull request being submitted to the DNN_Platform, if any of the following items are not true the pull request will be returned for edits referencing the specific missing item(s).
+
+* Change should represent an entire solution for the issue at hand.  Partial requests will NOT be processed.
+* Change should have a supporting issue logged on the DNN_Platform GitHub account, documenting the issue resolved, following the procedures outlined on the [Contribute Page](CONTRIBUTING.md)
+* Exception: Security items can be addressed individually, and should be initially communicated to the security@dnnsoftware.com email address for coordination
```
Comment (bdukes) [inline .github/PULL_REQUEST_PROCESS.md]: This line needs to be indented, and _Exception_ should probably be emphasized.

File: .github/PULL_REQUEST_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+* Pull request comment should contain at a minimum the following details (When creating, a default template will also prompt you for the proper information)
+--* Issue #, referenced such as “Fixes #24”
```
Comment (bdukes) [inline .github/PULL_REQUEST_PROCESS.md]: These indents aren't working.  To indent, just use spaces, e.g.

```
* Something
  * Something Else
* And another thing
```

File: .github/PULL_REQUEST_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+## Code Review Process
+Community review of submitted pull requests is encouraged, and all pull requests must be reviewed by at least **two (2)** community approvers before the change can be approved.  Once approved by two unique individuals the change can be merged by the secondary approval.  At the current time the following community members are designated approvers.
+
+* Mitchel Sellers ([mitchelsellers](https://github.com/mitchelsellers)) - Community Technology Advisory Group Lead
+* Oliver Hine ([ohine](https://github.com/ohine))
+* Brian Dukes ([bdukes](https://github.com/bdukes))
+* Peter Donker ([donker](https://github.com/donker)) - Community Developer Advisory Group Lead
+* Vicnec Masanas ([vmasanas](https://github.com/vmasanas))
+* Erik van Ballegoij ([erikvb](https://github.com/erikvb))
```
Comment (bdukes) [inline .github/PULL_REQUEST_PROCESS.md]: @tpluscode should be on this list, right?

File: .github/PULL_REQUEST_PROCESS.md (hunk 2 above)
Comment (bdukes) [reply .github/PULL_REQUEST_PROCESS.md]: How about:

 * Any tests related to the change have been updated
 * New tests covering the change are expected for areas of code already under test, and appreciated for areas of code not currently under test

File: .github/VERSIONING_POLICY.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+### Revision Releases (Ex 10.1.1, 10.1.2, 10.1.x)
+These releases are created primarily to contain hot-fix style improvements from prior releases.  Any bugs or security issues identified, or missing UI/UX features from a Minor/Major release might be added to a revision release.  Similar to a Minor release a Revision release will not contain any known breaking changes.
+## API Deprecation Policy
+The DNN Platform project is in a state of transition, continuing to modernize the API and work towards a transition to .NET Core.  To this point, it will be necessary for the project to remove public api's.  This will be done methodically, allowing developers to transition away from the older code with time to properly respond to change.
+Any API method to be removed will be flagged a deprecated in a release, major, minor or revision, and will be identified to be removed by a specific version.  This will be done using a C# annotation with a comment similar to the following "Deprecated in x.x.x, will be removed in y.0.0, use ____ instead".  The version number of "y" in this example must be 2 major versions ahead.
```
Comment (bdukes) [inline .github/VERSIONING_POLICY.md]: typo: "flagged _as_ deprecated"

File: .github/VERSIONING_POLICY.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+Major releases are also the time that platform requirements might be changed, such as requiring a new edition of SQL Server or otherwise.
+
+### Minor Releases (Ex 10.1, 10.2, 10.x)
+A minor might contain smaller new features and enhancements, but will not introduce any breaking API changes, nor will it change the requirements of the hosting environment or platform to run the application.  
+
+### Revision Releases (Ex 10.1.1, 10.1.2, 10.1.x)
+These releases are created primarily to contain hot-fix style improvements from prior releases.  Any bugs or security issues identified, or missing UI/UX features from a Minor/Major release might be added to a revision release.  Similar to a Minor release a Revision release will not contain any known breaking changes.
+## API Deprecation Policy
+The DNN Platform project is in a state of transition, continuing to modernize the API and work towards a transition to .NET Core.  To this point, it will be necessary for the project to remove public api's.  This will be done methodically, allowing developers to transition away from the older code with time to properly respond to change.
```
Comment (bdukes) [inline .github/VERSIONING_POLICY.md]: typo: change "api's" to "APIs"

**Comment by mitchelsellers** [comment]
I implemented the changes requested.  @bdukes  can you review/approve/merge?

File: .github/PULL_REQUEST_PROCESS.md (hunk 5 above)
Comment (bdukes) [reply .github/PULL_REQUEST_PROCESS.md]: @mitchelsellers can you please update this list of reviewers?

Comment (bdukes) [reply .github/PULL_REQUEST_PROCESS.md]: We need to add @iJungleboy

Comment (ohine) [reply .github/PULL_REQUEST_PROCESS.md]: That list is current for the community side of the approvers list. I'm going to merge the PR now. As we expand the @dnnsoftware/approvers group in the future we can add more people to the docs.

**Pull request merged**